### PR TITLE
Add SACD tensorboard

### LINKED ISSUE #488: TensorBoard logging in the SACD agent
Use the [TensorFlow SummaryWriter](https://www.tensorflow.org/api_docs/python/tf/summary/SummaryWriter) with `add_scalar` and `add_histogram` to log to TensorBoard when it's enabled.

### PR DESCRIPTION
**Pull request by Corentin-pro**
* Eager mode now disable/enable per agent

Fixes #488

### CODE REVIEW

File: ai/src/Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ test: venv
 	ALGORITHM="sacd" venv/bin/python3 -m pytest --timeout 30
 
 venv:
-	python3 -m venv venv
+	python3 -m venv -p 3.9 venv
```
Comment (lukekim) [inline ai/src/Makefile]: We need to stay on v3.8 for apple metal support.
```suggestion
	python3 -m venv venv
```

File: ai/src/train.py [hunk 2]
```diff
@@ -60,6 +60,7 @@ def __init__(
         self.log_dir = self.training_data_dir / "log"
 
         self.model_data_shape = data_manager.get_shape()
+        print('Getting agent')
```
Comment (lukekim) [inline ai/src/train.py]:
```suggestion
        print('Getting agent' + algorithm)
```

**Review by lukekim — CHANGES_REQUESTED** [review]

File: ai/src/Makefile (hunk 1 above)
Comment (phillipleblanc) [inline ai/src/Makefile]: We should keep using python 3.8 until we upgrade the Dockerfile for the project to use python 3.9 as well. As far as I know, we want to stay on python 3.8 for now as that will allow us to easily support Apple silicon: https://developer.apple.com/metal/tensorflow-plugin/

Comment (Corentin-pro) [reply ai/src/Makefile]: This was meant to run on my machine that runs 3.10, reverted

File: ai/src/train.py (hunk 2 above)
Comment (Corentin-pro) [reply ai/src/train.py]: That was a debug print, reverting

File: ai/src/algorithms/sacd/agent.py [hunk 3]
```diff
@@ -38,159 +37,149 @@ def create_network(
...
-                actor_tape.gradient(actor_loss, self._actor_variables),
-                self._actor_variables,
-            )
-        )
+                    tf.reduce_sum(action_probs * (self.alpha * action_logprobs - q_log_target), 1))
+            self._actor_optimizer.apply_gradients(
+                zip(actor_tape.gradient(actor_loss, self.actor.trainable_variables), self.actor.trainable_variables,))
 
-        with tf.name_scope("critic_loss"):
+            # with tf.name_scope("critic_loss"):
```
Comment (phillipleblanc) [inline ai/src/algorithms/sacd/agent.py:116]: Should this comment be removed?

File: ai/src/algorithms/sacd/agent.py [hunk 4]
```diff
@@ -38,159 +37,149 @@ def create_network(
...
-
-        with tf.name_scope("alpha_loss"):
+            self._critic_1_optimizer.apply_gradients(zip(
+                critic_tapes[0].gradient(critic_losses[0], self._critic_1.trainable_variables),
+                self._critic_1.trainable_variables,))
+            self._critic_2_optimizer.apply_gradients(zip(
+                critic_tapes[1].gradient(critic_losses[1], self._critic_2.trainable_variables),
+                self._critic_2.trainable_variables,))
+
+            # with tf.name_scope("alpha_loss"):
```
Comment (phillipleblanc) [inline ai/src/algorithms/sacd/agent.py:141]: Same for this one.

**Review by phillipleblanc — APPROVED** [review]

**Review by lukekim — APPROVED** [review]

**Pull request merged**